src/dos/drive_local.cpp

```diff
@@ -618,6 +640,23 @@ bool localFile::Write(uint8_t *data, uint16_t *size)
 		return true;
 	}
 
+	// Determine if the file is on a disk mounted as floppy or hard drive
+	const int type = local_drive.lock()->GetMediaByte();
+	if (type == 0xF0) {
```

Comment:
This mapping of media byte to disk type happens in two places. So there's an opportunity to extract this check into a static helper method that returns DiskType.